src/js/common/schemaform/FormPage.jsx

```diff
@@ -54,12 +55,16 @@ class FormPage extends React.Component {
   }
 
   onChange(formData) {
-    this.props.setData(formData);
+    let newData = formData;
+    if (this.props.route.pageConfig.showPagePerItem) {
+      newData = _.set([this.props.route.pageConfig.arrayPath, this.props.params.index], formData, this.props.form.data);
```

Comment:
It seems like you've explained most of what's going on here, in your summary comment from yesterday, and that a lot of what you've written could be repurposed to document this.